### DISCUSSION TITLE
Using base models with the timestep embedding?

### DISCUSSION THREAD

**Discussion by zaccharieramzi** [huggingface/diffusers#6664] [Q&A]
Hey!

I would like to reuse the models built here because I trust their implementation and would like to benefit from all the configuration choices.
However, I would like to do it for a use case that is not diffusion, does anyone know if it's possible to configure for example a Unet 1D model so that it doesn't rely on timestep embedding?

Thanks in advance!

**Comment by sayakpaul** [comment]
You can append the original implementation of UNet1D to not use timestep embedding.

**Reply by zaccharieramzi** [reply]
Hi @sayakpaul thanks for your answer

I am not sure what you mean by "append"? Do you mean subclass?

I have looked into it and the issue is that most of the blocks composing the Unet model are actually relying on timestep embeddings and it is goes down the pipe pretty far.

**Reply by sayakpaul** [reply]
Yeah that is what I meant. You can look for places where timestep is there and strategically remove them.
